packages/core/src/process/4-mutation-test-executor.ts

```diff
@@ -91,33 +91,76 @@ export class MutationTestExecutor {
+      };
+    }
+
+    const checkResult$ = new Subject<MutantResult>();
+    let previousPassedMutants$ = input$;
+
+    for (const checkerType of this.options.checkers) {
+      const passedMutants$ = new Subject<MutantTestCoverage>();
+      this.executeChecker(checkerType, previousPassedMutants$, checkResult$, passedMutants$);
+      previousPassedMutants$ = passedMutants$;
+    }
+
+    previousPassedMutants$.subscribe({
+      complete: () => {
+        this.log.debug('Checker(s) finished.');
+
+        checkResult$.complete();
+
+        this.checkerPool.dispose();
+        this.concurrencyTokenProvider.freeCheckers();
+      },
+    });
+
+    return {
+      checkResult$: checkResult$.asObservable(),
+      passedMutant$: previousPassedMutants$,
+    };
+  }
+
+  private async executeChecker(
+    checkerType: string,
+    previousPassedMutants$: Observable<MutantTestCoverage>,
+    checkResult$: Subject<MutantResult>,
+    passedMutant$: Subject<MutantTestCoverage>
+  ) {
+    const mutants = await lastValueFrom(merge(previousPassedMutants$).pipe(toArray()));
+
+    // Set the active checker on all checkerWorkers
+    await this.checkerPool.runOnAllResources(async (checkerResource) => {
+      await checkerResource.setActiveChecker(checkerType);
+    });
+
+    const groups = await firstValueFrom(
+      this.checkerPool.schedule(of(0), async (checker) => {
+        const group = await checker.createGroups?.(mutants);
+        return group ?? mutants.map((m) => [m]);
       })
-      .pipe(
-        // Dispose when all checks are completed.
-        // This will allow resources to be freed up and more test runners to be spined up.
-        tap({
-          complete: () => {
-            this.checkerPool.dispose();
-            this.concurrencyTokenProvider.freeCheckers();
-          },
-        }),
-        shareReplay()
-      );
-    const [passedCheckResult$, failedCheckResult$] = partition(checkTask$, ({ checkResult }) => checkResult.status === CheckStatus.Passed);
-    const checkResult$ = failedCheckResult$.pipe(
-      map((failedMutant) =>
-        this.mutationTestReportHelper.reportCheckFailed(failedMutant.mutant, failedMutant.checkResult as Exclude<CheckResult, PassedCheckResult>)
-      )
     );
-    const passedMutant$ = passedCheckResult$.pipe(map(({ mutant }) => mutant));
-    return { checkResult$, passedMutant$ };
+
+    this.log.debug(`${checkerType} created ${groups.length} groups.`);
+
+    const run$ = this.checkerPool.schedule(from(groups), async (checker, mutantGroup) => {
+      const results = await checker.check(mutantGroup);
+      results.forEach((result) => {
+        if (result.checkResult.status === CheckStatus.Passed) {
+          passedMutant$.next(result.mutant);
+        } else {
+          checkResult$.next(this.mutationTestReportHelper.reportCheckFailed(result.mutant, result.checkResult));
+        }
+      });
+    });
+
+    await lastValueFrom(run$);
+    passedMutant$.complete();
```

Comment:
I think we might be able to work without custom Subjects, instead work with pure RxJS pipes. That way we also don't need to manually `complete`.